File: Práctica1/java/src/deepspace/Loot.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author daniel
 */

// Representa el botín que se obtiene al vencer una nave
// enemiga

public class Loot {
    private int nSupplies;
    private int nWeapons;
    private int nShields;
    private int nHangars;
    private int nMedals;
    
    Loot(int nSup, int nWeap, int nSh, int nHan, int nMed){
        this.nSupplies = nSup;
        this.nWeapons = nWeap;
        this.nShields = nSh;
        this.nHangars = nHan;
        this.nMedals = nMed;        
    }
    
    int getNSupplies(){
        return nSupplies;
    }
    
    int getnWeapons(){
        return nWeapons;
    }
    
    int getnShields(){
        return nShields;
    }
    
    int getnHangars(){
        return nHangars;
    }
    
    int getnMedals(){
        return nMedals;
    }
}

File: Práctica2/lib/Loot.rb
#encoding: utf-8

# Representa el botín que se obtiene al vencer a una nave enemiga

module Deepspace
    class Loot
        def initialize(supplies, weapons, shields, hangars, medals)
            @nSupplies = supplies
            @nWeapons  = weapons
            @nShields  = shields
            @nHangars  = hangars
            @nMedals   = medals
        end

        def nSupplies
            @nSupplies
        end

        def nWeapons
            @nWeapons
        end

        def nShields
            @nShields
        end

        def nHangars
            @nHangars
        end

        def nMedals
            @nMedals
        end
        
    end
end

# Código de omprobación
#prueba = Deepspace::Loot.new(1,1,1,1,1)
#puts prueba.nHangars